libs/ojsonnet/Eval_jsonnet.ml

```diff
@@ -551,27 +539,33 @@ and eval_binary_op env el (op, tk) er =
  * return a cmp.
  *)
 and eval_std_cmp env tk (el : expr) (er : expr) : cmp =
-  let rec eval_std_cmp_value_ (v_el : V.value_) (v_er : V.value_) : cmp =
+  let rec eval_std_cmp_value_ (v_el : value_) (v_er : value_) : cmp =
     match (v_el, v_er) with
-    | V.Array (_, [||], _), V.Array (_, [||], _) -> Eq
-    | V.Array (_, [||], _), V.Array (_, _, _) -> Inf
-    | V.Array (_, _, _), V.Array (_, [||], _) -> Sup
-    | V.Array (al, ax, ar), V.Array (bl, bx, br) -> (
-        let a0 = Lazy.force ax.(0).v in
-        let b0 = Lazy.force bx.(0).v in
+    | Array (_, [||], _), Array (_, [||], _) -> Eq
+    | Array (_, [||], _), Array (_, _, _) -> Inf
+    | Array (_, _, _), Array (_, [||], _) -> Sup
+    | Array (al, ax, ar), Array (bl, bx, br) -> (
+        let a0 =
+          match ax.(0).value with
```

Comment:
You can probably factorize all those match Val | Unevaluated in a separate function like
eval_lazy_value so you don't have to repeat yourself all the time.